test/unit/app/utils/test_kenya_state_machine.py

```diff
@@ -153,10 +165,40 @@ def test_change_language(test_client, init_database, session_factory, user_input
 def test_opt_out_of_marketplace(test_client, init_database):
     session = opt_out_of_market_place_pin_authorization_state()
     user = standard_user()
-    #state_machine = KenyaUssdStateMachine(session, user)
+    # state_machine = KenyaUssdStateMachine(session, user)
 
-    #state_machine.feed_char("0000")
-    #assert state_machine.state == "complete"
-    #assert user.custom_attributes.market_enabled = False
+    # state_machine.feed_char("0000")
+    # assert state_machine.state == "complete"
+    # assert user.custom_attributes.market_enabled = False
     # TODO(owen): assert sms received is opt_out_of_market_place_sms
     return
+
+
+@pytest.mark.parametrize("session_factory, user_input, expected, failed_pin_attempts", [
+    (send_token_pin_authorization_state, "1212", "send_token_pin_authorization", 1)
```

Comment:
could combine this with `test_exit_pin_blocked`, making them different parameters. would also be good to to test User with two failed attempts: what happens if they input correct on third try, what happens if they input incorrect on third try

Reply:
Yeah sounds like a good idea, I'll add these with the next push.